### DISCUSSION TITLE
a record, filtering out itself for the related records M2M picker

### DISCUSSION THREAD

**Discussion by vahidreza** [directus/directus#13963] [Feature Requests]
this is not a bug but potentially a feature request. I just wanted to make sure if it exists or not.

I have an entity (namely student) which has an M2M relationship with itself (alias: friends). technically in the M2M interface, one should not be able to see himself but others.

in the filter section of the interface, how can i filter out the SELF record?
probably if the $SELF keywords existed i could easily use that as a reference to filter the result.

**Comment by azrikahar** [comment]
You can use the `$CURRENT_USER` dynamic variable (which should be what you're envisioning as `$SELF`): https://docs.directus.io/configuration/filter-rules/#dynamic-variables

> this is not a bug but potentially a feature request. I just wanted to make sure if it exists or not.

Feel free to use the discussions to ask questions or submit feature requests! The issues should be reserved for actual bugs with minimal reproducible steps.

**Reply by vahidreza** [reply]
@azrikahar tnx, but i dont mean to reference to the current user but i need to reference to the current record. so the user might be an admin but the record is a student's record.
as a better example consider a patient record which has related patients field. an admin (like a doctor) would set his related patients using the M2M interface but should not be able to select the main patient as one of the related patient records, the main patient should be left out using some sort of a filter.
so if we hade a `$SELF` keyword we would set it in a way that the patient id is not equal to `$SELF.id`

i hope i could convey my intention properly

**Reply by azrikahar** [reply]
Thanks for clarifying! Sounds pretty much like #12931. It's not currently supported, but could be a feature request indeed.

**Reply by vahidreza** [reply]
Any idea on when it might be on the development map? is there anywhere to upvote this feature?

**Reply by benhaynes** [reply]
@vahidreza — we only include broad features on our roadmap: https://directus.io/road-map/

Smaller updates are tracked here on GitHub Discussions... and we use the voting system above to track community desire.

**Reply by vahidreza** [reply]
thanks @benhaynes .
@azrikahar it would be great if you could point me to where one can start doing the mentioned change himself. like what files or methods i should look at to implement it maybe

**Reply by azrikahar** [reply]
@vahidreza currently the parsing of dynamic variables is happening here: https://github.com/directus/directus/blob/98ca14a950d32f41155f2c81181e72667ce123ae/packages/shared/src/utils/parse-filter.ts#L72-L92

However I'm not sure if this particular scenario it'll be as minimal as the other counterparts (since `value` may not have the field you are referring to). With that said, I've also just realized #2993 was already suggesting the same thing! Feel free to give it an upvote 👍

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
